src/components/profile/EditProfile.js

```diff
@@ -28,15 +28,28 @@ const EditProfile = ({ screenProps, theme, styles }) => {
     setProfile(storedProfile)
   }, [storedProfile])
 
+  const handleErrors = (errors, isValid) => {
+    setErrors(errors)
+    setIsValid(isValid)
+    return isValid
+  }
+
   const validate = debounce(async () => {
     log.info({ validate: profile })
+
     if (profile && profile.validate) {
       const { isValid, errors } = profile.validate()
+
+      // first we validate that the user-entered data is valid. If not, we inform the user about it
+      if (!isValid) {
```

Comment:
If there is an invalid username and an unavailable email the user will only get invalid values.
I suppose there is a trade-off on avoiding doing gun check if is invalid but I would prefer having all the feedback asap

Reply:
reverted